Add option to hide games from the library

arielj opened the pull request:
This PR adds a the ability to mark games from the library as `hidden` so they don't show up by default. This can be toggled on/off with the new checkbox.

Games can be hid/unhid using the right click context menu.

You can see it working in this video with the game A Short Hike (second to last, bottom right)

https://user-images.githubusercontent.com/188464/164943074-053065f1-bd62-4cd4-bc65-a77fa2344db6.mp4

(The total games counter changes by more than 1 because I have other hidden games outside the viewport)

This I tried and was not happy with (but accept any feedback):
- using an icon for the `Show hidden` toggle instead of text, it didn't look clear at least to me
- adding an icon in the game card to toggle the hide/unhide, it added noise for an action that's probably done only once or twice ever for a game

Haven't tested with Unreal Market (I don't use it, I don't know if this is expected to work there too? I understand it works but I don't know if we should ignore this feature there).

Closes #948

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

arielj commented on public/locales/bg/translation.json:
```diff
@@ -105,20 +105,17 @@
         "syncing": "Синхронизиране",
         "unsync": "Десинхронизиране"
     },
-    "Downloading": "Сваляне",
```
There are a lot of things being removed here by the `yarn i18next` command

arielj replied:
Oh I think it's because I removed that filter on a previous PR and didn't know I had to run `yarn i18next` again, those keys are not used anymore

flavioislima replied:
Yes

arielj commented on src/helpers/library.ts:
```diff
@@ -297,17 +297,17 @@ type RecentGame = {
+    (configStore.get('games.recent') as Array<RecentGame>) || []
+  const recentGamesList = recentGames.map((a) => a.appName) as string[]
+
+  return library.filter((game) => recentGamesList.includes(game.app_name))
```
While coding this feature initially I used a similar method like this one so I included this refactor. The refactor is a small improvement (we don't need to initialize the store and query it for every callback function call) but I can remove it from this PR if it adds noise, since it's not really needed for this feature at the end.

flavioislima commented:
Personally, I don't like the idea of having more text on the header.
I wonder if we could have another form control as we have for the platform filter and add the `Visibility` Icon there and change it to `VisibilityOFF` when the games are hidden.

something like this:
![image](https://user-images.githubusercontent.com/26871415/164971261-72bfaedf-eacd-4492-bc03-3daf9fc9cb91.png)

Using:
https://mui.com/material-ui/material-icons/?query=visibility&selected=Visibility
and
https://mui.com/material-ui/material-icons/?query=visibility&selected=VisibilityOff

what do you think?

arielj commented:
> Personally, I don't like the idea of having more text on the header. I wonder if we could have another form control as we have for the platform filter and add the `Visibility` Icon there and change it to `VisibilityOFF` when the games are hidden.
> 
> something like this: ![image](https://user-images.githubusercontent.com/26871415/164971261-72bfaedf-eacd-4492-bc03-3daf9fc9cb91.png)
> 
> Using: https://mui.com/material-ui/material-icons/?query=visibility&selected=Visibility and https://mui.com/material-ui/material-icons/?query=visibility&selected=VisibilityOff
> 
> what do you think?
> 
> with that approach, we can add a Star or Heart to show/hide favorites as well.

I can change that to try, what I'm worried about is that an eye icon doesn't make it obvious that it will show/hide hidden games. I imagine a user clicking the icon to see what happens, if they didn't hide any game nothing will happen.

I guess the tooltip would give a hint if the user leaves the mouse hovering the icon long enough, but tooltips don't show up when using the controller to navigate heroic.

Maybe I'm overthinking it xD

flavioislima commented:
Yes. Let's collect some feedback on the beta. 😀

arielj commented:
I moved this right next to the Total Games label, I imagine that would give more context that the icon is related to how many games you can see

![image](https://user-images.githubusercontent.com/188464/164983065-d042ec7d-bfa1-4334-a59b-a83dc9563d74.png)

I also fixed a bug, the selection of the filter was not persisted.

arielj commented on src/components/UI/FormControl/index.css:
```diff
@@ -93,6 +93,10 @@
 
+.FormControl--segmented > button > svg {
+  display: block;
+}
```
the svg icons where not properly sentered (svg images are inline by default and it was adding a few empty pixels under it)

flavioislima reviewed (approved):
Nice! :D
